test/tests_for_integration/api_to_db_test.py

```diff
@@ -924,3 +933,335 @@ def _run_job_concierge_fail(
+            "scheduler_id": "123",
+            "scheduler_type": "condor",
+        }
+        assert job1 == expected_job1
+
+        expected_job2 = {
+            "_id": ObjectId(job_id_2),
+            "user": USER_NO_ADMIN,
+            "authstrat": "kbaseworkspace",
+            "wsid": 1,
+            "status": "queued",
+            "job_input": {
+                "wsid": 1,
+                "method": "mod2.meth2",
+                "params": [{"baz": "bat"}, 3.14],
+                "service_ver": "somehash2",
+                "app_id": "mod2/app2",
+                "source_ws_objects": [],
+                "parent_job_id": parent_job_id,
+                "requirements": {
+                    "clientgroup": "bigmem",
+                    "cpu": 4,
+                    "memory": 2000,
+                    "disk": 100,
+                },
+                "narrative_cell_info": {},
+            },
+            "child_jobs": [],
+            "batch_job": False,
+            "scheduler_id": "456",
+            "scheduler_type": "condor",
+        }
+        assert job2 == expected_job2
+
+        parent_job = _get_mongo_job(mongo_client, parent_job_id, has_queued=False)
+        expected_parent_job = {
+            "_id": ObjectId(parent_job_id),
+            "user": USER_NO_ADMIN,
+            "authstrat": "kbaseworkspace",
+            "wsid": 2,
+            "status": "created",
+            "job_input": {
+                "method": "batch",
+                "service_ver": "batch",
+                "app_id": "batch",
+                "source_ws_objects": [],
+                "narrative_cell_info": {
+                    "run_id": "rid2",
+                    "token_id": "tid2",
+                    "tag": "yourit2",
+                    "cell_id": "cid2",
+                },
+            },
+            "child_jobs": [job_id_1, job_id_2],
+            "batch_job": True,
+        }
+        assert parent_job == expected_parent_job
+
+        expected_sub_1 = _get_condor_sub_for_rj_param_set(
+            job_id_1,
+            USER_NO_ADMIN,
+            TOKEN_NO_ADMIN,
+            clientgroup="njs",
+            cpu=8,
+            mem=5,
+            disk=30,
+            parent_job_id=parent_job_id,
+        )
+        expected_sub_1["+KB_WSID"] = ""
+        expected_sub_2 = _get_condor_sub_for_rj_param_set(
+            job_id_2,
+            USER_NO_ADMIN,
+            TOKEN_NO_ADMIN,
+            clientgroup="bigmem",
+            cpu=4,
+            mem=2000,
+            disk=100,
+            parent_job_id=parent_job_id,
+        )
+        expected_sub_2.update(
+            {
+                "+KB_MODULE_NAME": '"mod2"',
+                "+KB_FUNCTION_NAME": '"meth2"',
+                "+KB_APP_ID": '"mod2/app2"',
+                "+KB_APP_MODULE_NAME": '"mod2"',
+                "+KB_SOURCE_WS_OBJECTS": "",
+            }
+        )
+
+        assert sub_init.call_args_list == [call(expected_sub_1), call(expected_sub_2)]
```

Comment:
The tests look reasonable to me. Not sure what's going on with the mocked objects in here, but I wonder if knowing that the HTC Condor objects are supposed to be doing things with threads would help.

Also, are we OK with the tradition of very long tests?

Reply:
> I wonder if knowing that the HTC Condor objects are supposed to be doing things with threads would help.

I'm not 100% sure, but I don't think so. For some reason the mocking lib is including subcalls to the mocks in the list of calls for a particular mock, which I don't think can be due to threading, because the Condor code shouldn't be called at all since it's all mocked out.

Reply:
> Also, are we OK with the tradition of very long tests?

I mean, I prefer short tests, but in this case it's pretty hard to shorten it up without doing kind of pointless things like making methods for creating each argument and response set. A large portion of the length is due to large data structures that are being checked - the actual LOC is probably reasonable, but the test looks huge due to the inline data structures.

Reply:
Personally, I'd rather just have the data structures in line than have to jump around to various functions to see what they are *shrug*

Reply:
And finally, integration tests are often pretty hairy, which is one reason to keep them to a minimum